here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/HttpStorageReadExecute.java

```diff
@@ -82,55 +81,56 @@ private static Result executeFeaturesById(
+    return prepareResult(response);
   }
 
-  private static Result executeFeatureByBBox(
+  private static Response executeFeatureByBBox(
       @NotNull NakshaContext context, ReadFeaturesProxyWrapper readRequest, RequestSender requestSender) {
     String queryParamsString = keysToKeyValuesStrings(readRequest, WEST, NORTH, EAST, SOUTH, LIMIT);
 
     HttpResponse<byte[]> response = requestSender.sendRequest(
         format("/%s/bbox?%s%s", baseEndpoint(readRequest), queryParamsString, getPOpQueryOrEmpty(readRequest)),
         Map.of(HDR_STREAM_ID, context.getStreamId()));
 
-    return prepareResult(response, XyzFeatureCollection.class, XyzFeatureCollection::getFeatures);
+    return prepareResult(response);
   }
 
-  private static Result executeFeaturesByTile(
+  private static Response executeFeaturesByTile(
       @NotNull NakshaContext context, ReadFeaturesProxyWrapper readRequest, RequestSender requestSender) {
     String queryParamsString = keysToKeyValuesStrings(readRequest, MARGIN, LIMIT);
     String tileType = readRequest.getQueryParameter(TILE_TYPE);
     String tileId = readRequest.getQueryParameter(TILE_ID);
 
     if (tileType != null && !tileType.equals(TILE_TYPE_QUADKEY))
-      return new ErrorResult(XyzError.NOT_IMPLEMENTED, "Tile type other than " + TILE_TYPE_QUADKEY);
+      return new ErrorResponse(new NakshaError(
+          NakshaError.NOT_IMPLEMENTED, "Tile type other than " + TILE_TYPE_QUADKEY, null, null));
 
     HttpResponse<byte[]> response = requestSender.sendRequest(
         format(
             "/%s/quadkey/%s?%s%s",
             baseEndpoint(readRequest), tileId, queryParamsString, getPOpQueryOrEmpty(readRequest)),
         Map.of(HDR_STREAM_ID, context.getStreamId()));
 
-    return prepareResult(response, XyzFeatureCollection.class, XyzFeatureCollection::getFeatures);
+    return prepareResult(response);
   }
 
-  private static Result executeIterate(
+  private static Response executeIterate(
       @NotNull NakshaContext context, ReadFeaturesProxyWrapper readRequest, RequestSender requestSender) {
     String queryParamsString = keysToKeyValuesStrings(readRequest, LIMIT);
 
     HttpResponse<byte[]> response = requestSender.sendRequest(
         format("/%s/iterate?%s", baseEndpoint(readRequest), queryParamsString),
         Map.of(HDR_STREAM_ID, context.getStreamId()));
 
-    return prepareResult(response, XyzFeatureCollection.class, XyzFeatureCollection::getFeatures);
+    return prepareResult(response);
   }
 
   /**
    * @return either POp query string starting with "&" or an empty string if the POp is null
    */
   private static String getPOpQueryOrEmpty(ReadFeaturesProxyWrapper readRequest) {
-    POp pOp = readRequest.getQueryParameter(PROPERTY_SEARCH_OP);
-    return pOp == null ? "" : "&" + POpToQueryConverter.p0pToQuery(pOp);
+    final IPropertyQuery propertyQuery = readRequest.getQuery().getProperties();
+    return propertyQuery == null ? "" : "&" + propertyQuery;
```

Comment:
It would be nice to test it with
`ReadFeaturesByBBoxHttpStorageTest.tc800_testPropertySearch()`
If the test passes that means that whole POp conversion and its test can be removed from HttpStorage

However I was not able to run the test due to compilation errors in `AbstractEventHandler` and  other classes

Reply:
Fixing that test will be part of fixing compiler errors of lib-handler. But rest assured that POp related classes are now deprecated.